pkg/platform/kube/functionres/lazy.go

```diff
@@ -2184,23 +2191,86 @@ func (lc *lazyClient) resolveFailFast(pods []v1.Pod,
 				}
 			}
 		}
+	}
 
+	// check each pod's conditions to determine if there is at least one unschedulable pod
+	errGroup, _ := errgroup.WithContext(context.TODO(), lc.logger)
+	lock := sync.Mutex{}
+	scaleUpOccurred := false
+	for _, pod := range pods {
+		pod := pod
 		for _, condition := range pod.Status.Conditions {
 
 			// check if the pod is in pending state, and the reason is that it is unschedulable
 			// (meaning no k8s node can currently run it, because of insufficient resources etc..)
 			if pod.Status.Phase == v1.PodPending && condition.Reason == "Unschedulable" {
 
-				// TODO: if pods has an event that cluster is going to scale up - don't return error
-
-				return errors.Errorf("NuclioFunction pod (%s) is unschedulable", pod.Name)
+				errGroup.Go("WaitAndCheckAutoScaleEvents", func() error {
+
+					// sleeping for 15s because autoscale cycle is 10s minimum
+					lc.logger.Debug("Waiting 15 seconds for autoscale evaluation")
+					time.Sleep(15 * time.Second)
+
+					// check if the pod is unschedulable due to scaling up
+					triggeredScaleUp, err := lc.isPodAutoScaledUp(pod)
+					if err != nil {
+						return errors.Wrapf(err,
+							"Failed to resolve pod (%s) triggered a node scale up",
+							pod.Name)
```

Comment:
lets not "trigger" the fail fast in case of IO errors against kubernetes. maybe the get pod events failed transiently. we can let the fail-fast do another iteration and once we successfully infer whether a pod is unschedulable - then "trigger" the fail-fast. if we hit some transient errors or so, just let the fail-fast run again